gemini/sample-apps/customer-search/functions/create_fd/main.py

```diff
@@ -0,0 +1,230 @@
+"""This is a python utility file."""
+
+# pylint: disable=R0801
+# pylint: disable=R0914
+
+import datetime
+from datetime import date
+from os import environ
+import random
+import re
+
+import functions_framework
+from google.cloud import bigquery
+from utils.bq_query_handler import BigQueryHandler
+from utils.gemini import Gemini
+
+project_id = environ.get("PROJECT_ID")
+
+
+def get_ac_number():
+    """
+    Generates a random account number.
+
+    Returns:
+        A random account number as an integer.
+    """
+
+    account_number = "11000"
+    for _ in range(5):
+        account_number += str(random.randint(0, 9))
+
+    return int(account_number)
+
+
+def check_senior_citizen(dob):
+    """
+    Checks whether a person is a senior citizen or not using their date of birth.
+
+    Args:
+      dob: A datetime.date object representing the person's date of birth.
+
+    Returns:
+      True if the person is a senior citizen, False otherwise.
+    """
+
+    today = datetime.date.today()
+    age = today.year - dob.year - ((today.month, today.day) < (dob.month, dob.day))
+    return age >= 60
+
+
+def get_number_of_days(fd_tenure):
+    """
+    Calculates the number of days in a fixed deposit tenure.
+
+    Args:
+      fd_tenure: A string representing the fixed deposit tenure.
+
+    Returns:
+      The number of days in the fixed deposit tenure as an integer.
+    """
+
+    years = 0
+    months = 0
+    days = 0
+
+    if re.search(r"(\d+) year", fd_tenure) is not None:
+        years = re.search(r"(\d+) year", fd_tenure).group(1)
+    if re.search(r"(\d+) month", fd_tenure) is not None:
+        months = re.search(r"(\d+) month", fd_tenure).group(1)
+    if re.search(r"(\d+) day", fd_tenure) is not None:
+        days = re.search(r"(\d+) day", fd_tenure).group(1)
+
+    # Convert the years, months, and days to integers.
+    years = int(years)
+    months = int(months)
+    days = int(days)
+
+    # Calculate the total number of days in the tenure.
+    total_days = 365 * years + 30 * months + days
+
+    return total_days
```

Comment:
Depending on the structure of the input string, this method can likely be simplified into one regex